chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactionlist/ui/screen/TransactionListScreen.kt

```diff
@@ -0,0 +1,171 @@
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.gojek.chuckmqtt.R
+import com.gojek.chuckmqtt.internal.presentation.model.MqttTransactionUiModel
+import com.gojek.chuckmqtt.internal.presentation.theme.ChuckMqttTheme
+import com.gojek.chuckmqtt.internal.presentation.transactionlist.mvi.TransactionListIntent
+import com.gojek.chuckmqtt.internal.presentation.transactionlist.mvi.TransactionListIntent.ClearTransactionHistoryIntent
+import com.gojek.chuckmqtt.internal.presentation.transactionlist.mvi.TransactionListIntent.OpenTransactionDetailIntent
+import com.gojek.chuckmqtt.internal.presentation.transactionlist.mvi.TransactionListIntent.StartObservingAllTransactionsIntent
+import com.gojek.chuckmqtt.internal.presentation.transactionlist.mvi.TransactionListViewState
+import com.gojek.chuckmqtt.internal.presentation.transactionlist.viewmodel.TransactionListViewModel
+
+@Composable
+internal fun TransactionListScreen(
+    toolbarSubtitle: CharSequence,
+    viewModel: TransactionListViewModel,
+    intentLambda: (TransactionListIntent) -> Unit
+) {
+    val state by viewModel.states().subscribeAsState(TransactionListViewState.default())
+
+    ChuckMqttTheme {
+        Scaffold(
+            topBar = { ListAppBar(toolbarSubtitle = toolbarSubtitle, intentLambda) },
+            content = {
+                TransactionList(state.transactionList, intentLambda)
+
+                LaunchedEffect(Unit) {
+                    intentLambda(StartObservingAllTransactionsIntent)
+                }
+            }
+        )
+    }
+}
+
+@Composable
+internal fun ListAppBar(
+    toolbarSubtitle: CharSequence,
+    intentLambda: (TransactionListIntent) -> Unit
+) {
+    TopAppBar(
+        title = {
+            Column {
+                Text("MQTT Chuck")
+                Text(
+                    text = toolbarSubtitle.toString(),
+                    style = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Normal)
+                )
+            }
+        },
+        actions = {
+            IconButton(onClick = { intentLambda(ClearTransactionHistoryIntent) }) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mqtt_chuck_ic_delete_white_24dp),
+                    contentDescription = "Delete Packets"
+                )
+            }
+        },
+        backgroundColor = MaterialTheme.colors.primarySurface,
+        elevation = 4.dp
+    )
+}
+
+@Composable
+internal fun TransactionList(
+    transactionList: List<MqttTransactionUiModel>,
+    intentLambda: (TransactionListIntent) -> Unit
+) {
+    LazyColumn {
+        items(transactionList) { item ->
+            TransactionListItem(item = item, intentLambda = intentLambda)
+            Divider(color = Color.LightGray)
```

Comment:
Let's use colors through the theme to support dark mode, Similarly, all the other places where colors and icons which are dependent on theme